File: src/main/java/com/mpe85/grampa/rule/Action.java
package com.mpe85.grampa.rule;

/**
 * A parser action
 * 
 * @author mpe85
 *
 * @param <T>
 *            the type of the stack elements
 */
@FunctionalInterface
public interface Action<T> {
	
	/**
	 * Runs the action.
	 * 
	 * @param context
	 *            an action context
	 * @return true if the action was successfully run, false otherwise
	 */
	boolean run(ActionContext<T> context);
	
}

File: src/main/kotlin/com/mpe85/grampa/rule/Action.kt
package com.mpe85.grampa.rule

import com.mpe85.grampa.context.RuleContext

/**
 * A parser action.
 *
 * @author mpe85
 * @param[T] the type of the stack elements
 */
@FunctionalInterface
fun interface Action<T> {

    /**
     * Run the action.
     *
     * @param [context] an action context
     * @return true if the action was successfully run
     */
    fun run(context: RuleContext<T>): Boolean

}
